templates/common/aws/units/afterburn-hostname.service.yaml

```diff
@@ -0,0 +1,18 @@
+name: afterburn-hostname.service
+enabled: true
+contents: |
+  [Unit]
+  Description=Afterburn Hostname for AWS
+  After=NetworkManager-wait-online.service
+  Before=node-valid-hostname.service kubelet.service
+
+  [Service]
+  Restart=on-failure
+  RestartSec=15
+  Type=oneshot
+  RemainAfterExit=yes
+  ExecStartPre=/usr/bin/afterburn --provider aws --hostname=/run/afterburn.hostname
+  ExecStart=/bin/bash -c "source /usr/local/sbin/set-valid-hostname.sh; set_valid_hostname `cat /run/afterburn.hostname`"
```

Comment:
This will need reworking to be based on https://github.com/openshift/machine-config-operator/pull/2618